command/git.go

```diff
@@ -147,16 +150,15 @@ func (opts cloneOpts) buildHTTPCloneCommand() ([]string, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to parse URL from location")
 	}
-	clone := fmt.Sprintf("git clone https://%s@%s/%s/%s.git '%s'", opts.token, urlLocation.Host, opts.owner, opts.repo, opts.dir)
-	if opts.branch != "" {
-		clone = fmt.Sprintf("%s --branch '%s'", clone, opts.branch)
-	}
+	cloneCmd := fmt.Sprintf("git clone https://%s@%s/%s/%s.git '%s'", opts.token, urlLocation.Host, opts.owner, opts.repo, opts.dir)
+	cloneCmd = fmt.Sprintf("%s --branch '%s'", cloneCmd, opts.branch)
+	cloneCmd = fmt.Sprintf("%s --depth %d", cloneCmd, defaultCloneDepth)
 
-	redactedClone := strings.Replace(clone, opts.token, "[redacted oauth token]", -1)
+	redactedCloneCmd := strings.Replace(cloneCmd, opts.token, "[redacted oauth token]", -1)
 	return []string{
 		"set +o xtrace",
-		fmt.Sprintf(`echo %s`, strconv.Quote(redactedClone)),
-		clone,
+		fmt.Sprintf(`echo %s`, strconv.Quote(redactedCloneCmd)),
+		cloneCmd,
```

Comment:
the xtrace state feels wrong.

Reply:
also we should probably pipefail

Reply:
I think the point of the xtrace is to turn it off when we echo. Seems like we could remove that altogether though. Removed.